Src/BootCamp.Chapter/Checks.cs

```diff
@@ -1,39 +1,214 @@
+                    if (personCount == 1) // the second account to be added
+                    {
+                        name.Append(" and " + account.GetName());
+                    }
+                    else // any others
+                    {
+                        name.Replace(" and ", ", ");
+                        name.Append(" and " + account.GetName());
+                    }
+                    personCount++;
+                }
+                else if (amount > account.CurrentBalance()) // this is now the poorest account.
+                {
+                    name.Clear();
+                    name.Append(account.GetName());
+                    amount = account.CurrentBalance();
+                    personCount = 1;
+                }
+            }
         }
 
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
-        }
+            if (Testers.IsThisStringArrayValid(peopleAndBalances))
+            {
+                return nA;
+            }
+            StringBuilder name = new StringBuilder();
+            decimal amount = 0;
+            int personCount = 1;
 
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                Account account = AccountCreator(peopleAndBalances[i]);
+
+                if (i == 0)
+                {
+                    name.Append(account.GetName());
+                    amount = account.CurrentBalance();
+                }
+                else if (amount == account.CurrentBalance())
+                {
+                    if (personCount == 1)
+                    {
+                        name.Append(" and " + account.GetName());
+                    }
+                    else
+                    {
+                        name.Replace(" and ", ", ");
+                        name.Append(" and " + account.GetName());
+                    }
+                    personCount++;
+                }
+                else if (amount < account.CurrentBalance())
+                {
+                    name.Clear();
+                    name.Append(account.GetName());
+                    amount = account.CurrentBalance();
+                    personCount = 1;
+                }
+            }
+
+            return MakeStuffWithInput.MakefullstringRichestPeople(personCount, name.ToString(), amount);
+        }
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (Testers.IsThisStringArrayValid(peopleAndBalances))
+            {
+                return nA;
+            }
+            StringBuilder name = new StringBuilder();
+            decimal amount = 0;
+            int personCount = 1;
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                Account account = AccountCreator(peopleAndBalances[i]);
+
+                if (account.IsLessThan2Balance())
+                {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Either remove or fill this block of code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5123803)